yopt/src/main/scala/pravda/yopt/CmdDecoder.scala

```diff
@@ -26,22 +30,22 @@ trait CmdDecoder[T] {
 
 object CmdDecoder {
   private val noValueError = Left("Option must have value. No value provided")
-  private def parseError(ex: NumberFormatException) =
-    Left(s"Error parsing option value: ${ex.getClass}:${ex.getMessage}")
-
+  
   implicit val intDecoder: CmdDecoder[Int] = new CmdDecoder[Int] {
     override def decode(line: Line): Either[String, (Int, Line)] =
       line.headOption
         .map { item =>
-          try {
-            Right((Integer.parseInt(item), line.tail))
-          } catch {
-            case ex: NumberFormatException =>
-              try {
-                Right((Integer.parseInt(item.drop(2), 16), line.tail))
-              } catch {
-                case ex: NumberFormatException => parseError(ex)
-              }
+          item.toLowerCase match {
+            case neg if neg.head == '-' => Left(NegativeNumberDecodeError.toString)
+            case hex if hex.startsWith("0x") =>
```

Comment:
I don't think it's a good solution to always parse hex numbers when user specifies `Int`. It looks even more confusing taking into account the case above. I can write `0xffffffff` what will be parsed as `-1`.

Reply:
The user's input will be parsed as a hex number only if the user intentionally has typed "0x". Otherwise, it will be considered as a number in decimal format. The attempt to parse the user's input as a hex number have been presented in the existing code already. So I have decided to retain it since I had no enough requirements to make another decision.

If you have another vision please describe it, I'll change the code.

Reply:
The best solution here is to implement parser for tagged type like `Int @@ Hex`. See https://github.com/rudogma/scala-supertagged , we use them also as the core dependency.

Reply:
Or just remove the hex number parsing :D

Reply:
I have added two new types and corresponding parsers: Hex (tagged Long) and YesOrNo (tagged Boolean). The hexadecimal parser was removed from Int and Long types, also "yes" and "no" options were removed from the Boolean parser.